src/frontend/src/components/FilterPanel/components/QCStatusFilter/index.tsx

```diff
@@ -0,0 +1,71 @@
+import { DefaultMenuSelectOption } from "czifui";
+import { useEffect, useState } from "react";
+import { StyledComplexFilter } from "../../style";
+
+interface Props {
+  options: DefaultMenuSelectOption[];
+  updateQCStatusFilter: (qcStatuses: string[]) => void;
+}
+
+export type ComplexFilterValue =
+  | DefaultMenuSelectOption[]
+  | DefaultMenuSelectOption
+  | null;
+
+// ComplexFilter defaults to checking if option is selected (is a value) by equality.
+// However, because our options are objects that are dynamically generated, object
+// equality is always false because they're not the same object, even if same content.
+// Instead, we must create a custom selection checker to check underlying value.
```

Comment:
I don't think this is needed anymore since an SDS update.

Reply:
(not that high of priority since I'm pretty sure SDS just implemented this exact thing, there aren't any performance improvements etc.)

Reply:
ok cool, good to know!